### DISCUSSION TITLE
Issue reach /v2/ api on a self hosted zitadel

### DISCUSSION THREAD

**Discussion by salvatorebic** [zitadel/zitadel#8794] [Q&A] [answered]
Hi,

i installed zitadel on a docker container with caddy as proxy (used netbird quick guide for self hosting), now i cannot interact with any /v2/ api , i always get a 404 error message. I added the /v2/ path on my Caddyfile .

Example of url i am trying to use https://custom-domain/v2/users/:myuserid/links .

I have no problem using /management/v1/* api .

Am i missing somenthing?

Thank you.

**Comment by fforootd** [comment]
Can you share your caddy file and the full response you get (maybe also caddy logs)?

**Comment by salvatorebic** [comment]
sure, 

here the POST request

![image](https://github.com/user-attachments/assets/79db28b7-5db9-4100-b628-db964a17f90e)

and caddy file

[Caddyfile.txt](https://github.com/user-attachments/files/17432183/Caddyfile.txt)

**Reply by fforootd** [reply]
Can you try what happens if you remove the : in front of the userid?

**Reply by salvatorebic** [reply]
Same results :

![image](https://github.com/user-attachments/assets/0814359c-8137-4cee-9d9e-7ea80afce7cd)

I have already checked out the ids of user and idp to be sure

**Reply by fforootd** [reply]
Hm weird, a get on the user does work?

https://zitadel.com/docs/apis/resources/user_service_v2/user-service-get-user-by-id

**Reply by salvatorebic** [reply]
No, tried that one too thinking i was making some mistake on the POST request. Seems  like any v2 API are not responding.

**Reply by fforootd** [reply]
What version of zitadel are you using? Old version did not contain the v2 api 😁

**Reply by salvatorebic** [reply]
i thought that too but i cannot find where is indicated the version installed ...

**Reply by salvatorebic** [reply]
nevermind , just found it , v2.54.3

**Reply by fforootd** [reply]
Zitadel prints the version at startup. Not sure how you deployed zitadel 😁

**Reply by fforootd** [reply, marked as answer]
https://github.com/zitadel/zitadel/releases/tag/v2.58.0 this graduated the v2 to stable

so you might want to try v2beta as a path in your version

**Reply by salvatorebic** [reply]
A foolish mistake by my part , i will install a more recent version, thank you very much .

**Reply by fforootd** [reply]
No worries, happy to help!
